### ISSUE TITLE
Allow Conditional Loading of Modules per Server

### ISSUE THREAD

**Issue by IanMitchell** [IanMitchell/aquarius#43]
When Aquarius is added to a server, only a select few modules should be enabled automatically. The rest will need to be enabled by the server admin.

As part of the settings, Aquarius should also allow the restriction of modules, perhaps using the following states:

1. **ON**: Anyone can use this module
2. **RESTRICTED**: Only the `Aquarius Mod` role and server admin can use this module
3. **ADMIN**: Only the server admin can use this module
4. **OFF**: The module won't run on the server

Further, additional configuration values should be allowable (things like API access keys and the like)

**Comment by IanMitchell** [comment]
Some additional notes - we'll probably need a new messaging utility. Something like 

```
prompt('input string').then(response => ... ).catch(err => ...);
```

Which queries the server admin and resolves with their response. The user can respond `cancel` to stop the process, which throws the error that the module keys off of to clean up any activity.

We'll also want an optional `setUp` method per command that can prompt for keys or other such information. This can be used as an temporary whitelist/blacklist system as well, until a better one can be devised (just exporting an array of IDs?)

**Comment by IanMitchell** [comment]
The setup method should probably be a promise as well; the core bot can key off of a resolve to finish adding the module or a reject to send an error message. At this point, it might be prudent to switch the command files to a class system in a similar fashion to ARX-7

**Comment by IanMitchell** [comment]
Once this is done, we should bump up the version to 1.0 and release it on NPM

**Comment by IanMitchell** [comment]
Thinking the following format might be best - maybe rename `setUp` to `addServer`?

```javascript
class Command {
  constructor() {
    this.name = 'Command';
    this.serverWhitelist = [];
    this.serverBlacklist = [];
  }

  setUp() {
    return;
  }

  message() {
    return false;
  }

  helpMessage() {
    return;
  }
}

module.exports = Command;
```

**Comment by IanMitchell** [comment]
As part of this, we need a "settings" table - the Schema should probably look like:

id  | serverID  | command  | key  | value
--- | ------------ | -------------- | ------ | ------- |
x | '3423...' | 'karma' | 'name' | 'torneritos' |

**Comment by IanMitchell** [comment]
Actually, it might be worth combining key/value into a single `config` JSON field

**Comment by IanMitchell** [comment]
I think this might require a few extra database tables; it might be worth reorganizing what we currently have.

**Comment by IanMitchell** [comment]
Progress update: Settings and conditional command loading work. I've gotten an initial admin query system up that allows them to set config values, but I still need to add/remove all commands, clear up the messaging a bit, and clear set config values.

**Comment by IanMitchell** [comment]
Alright, status update - everything is done except for:

* Update Karma to read from settings
* New server created message
* Create admin command to automatically create Aquarius Roles

**Comment by IanMitchell** [comment]
Issues found:

- [ ] After adding all commands, a `help` displays an empty list of available commands
- [ ] Karma is broken

**Comment by IanMitchell** [comment]
Done and merged in 8256f76e25d72c3578b99def0cf7a3fccc15fd9d
